app/_src/kubernetes-ingress-controller/guides/using-gateway-discovery.md

```diff
@@ -0,0 +1,246 @@
+
+```
+$ kubectl get deployment -n kong
+NAME              READY   UP-TO-DATE   AVAILABLE   AGE
+controller-kong   3/3     3            3           5m
+gateway-kong      4/4     4            4           5m
+```
+
+## Testing configuration
+
+In order to verify that the configuration is indeed sent to all the Gateways we
+can deploy a simple HTTP Service:
+
+```
+echo "
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: httpbin
+  labels:
+    app: httpbin
+spec:
+  selector:
+    matchLabels:
+      app: httpbin
+  template:
+    metadata:
+      labels:
+        app: httpbin
+    spec:
+      containers:
+      - name: httpbin
+        image: kong/httpbin:0.1.0
+        ports:
+        - containerPort: 80
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: httpbin
+  name: httpbin
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: httpbin
+  type: ClusterIP
+" | kubectl apply -f -
+```
+
+with an `Ingress` with an `IngressClass`:
+
+```
+echo "
+apiVersion: networking.k8s.io/v1
+kind: IngressClass
+metadata:
+  name: kong
+spec:
+  controller: ingress-controllers.konghq.com/kong
+ ---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: echo
+  annotations:
+    konghq.com/strip-path: 'true'
+spec:
+  ingressClassName: kong
+  rules:
+  - host: kong.example
+    http:
+      paths:
+      - path: /echo
+        pathType: Prefix
+        backend:
+          service:
+            name: httpbin
+            port:
+              number: 80
+" | kubectl apply -f -
+```
+
+With those manifests applied we can observe controller logs for entries that
+indicate successful configuration of all the discovered Gateways:
+
+```
+$ stern -n kong -lapp=controller-kong --since 1m --include "synced configuration"
+...
+controller-kong-545d798874-q6h7m ingress-controller time="2023-03-02T15:11:42Z" level=info msg="successfully synced configuration to kong" kong_url="https://10.244.0.29:8444"
+controller-kong-545d798874-q6h7m ingress-controller time="2023-03-02T15:11:42Z" level=info msg="successfully synced configuration to kong" kong_url="https://10.244.0.15:8444"
+controller-kong-545d798874-q6h7m ingress-controller time="2023-03-02T15:11:42Z" level=info msg="successfully synced configuration to kong" kong_url="https://10.244.0.16:8444"
+controller-kong-545d798874-q6h7m ingress-controller time="2023-03-02T15:11:42Z" level=info msg="successfully synced configuration to kong" kong_url="https://10.244.0.30:8444"
+```
+
+At this point we should be able to access the `/echo` endpoint from our `htptbin` service:
+
+```
+$ curl -i http://kong.example/echo --resolve kong.example:80:$PROXY_IP
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+...
+```
+
+After issuing several queries against that address we can see in Gateway logs
+that we're hitting all Pod(s) which proxy traffic as configured (mark the first
+column: pod name):
```

Comment:
Maybe to be explicit, we could mention here that the traffic between Gateway pods is load-balanced by k8s service / kube-proxy?

Reply:
Added this mention. PTAL